coderd/util/slice/slice.go

```diff
@@ -230,3 +230,13 @@ func Convert[F any, T any](a []F, f func(F) T) []T {
 	}
 	return tmp
 }
+
+func ToMap[T any, K comparable, V any](a []T, cnv func(t T) (K, V)) map[K]V {
+	m := make(map[K]V, len(a))
+
+	for i := range a {
+		k, v := cnv(a[i])
+		m[k] = v
+	}
+	return m
+}
```

Comment:
nit: `ToMapFunc` as I would assume `slices.ToMap` would take a variadic list of key-value pairs, e.g.

```
slices.ToMap('a', 1, 'b', 2) => map[string]int{'a': 1, 'b': 2}
```